File: app/src/main/java/vn/loitp/app/activity/function/hashmap/HashMapActivity.java
package vn.loitp.app.activity.function.hashmap;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.core.base.BaseFontActivity;

import java.util.HashMap;
import java.util.Map;

import loitp.basemaster.R;

public class HashMapActivity extends BaseFontActivity implements View.OnClickListener {
    private TextView tv;
    private Map<String, String> map = new HashMap<String, String>();

    private int autoKey = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        tv = (TextView) findViewById(R.id.tv);
        findViewById(R.id.bt_add).setOnClickListener(this);
        findViewById(R.id.bt_get_key_0).setOnClickListener(this);
        findViewById(R.id.bt_remove_key_0).setOnClickListener(this);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_hashmap;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.bt_add:
                map.put(String.valueOf(autoKey), "Value " + autoKey);
                printMap();
                autoKey++;
                break;
            case R.id.bt_get_key_0:
                String value = map.get(String.valueOf(0));
                showShort("Click value= " + value);
                break;
            case R.id.bt_remove_key_0:
                map.remove(String.valueOf(0));
                printMap();
                break;
        }
    }

    private void printMap() {
        String s = "";
        for (Map.Entry<String, String> entry : map.entrySet()) {
            s += entry.getKey() + " -> " + entry.getValue() + "\n";
        }
        tv.setText(s.isEmpty() ? "No data" : s);
    }
}

File: app/src/main/java/vn/loitp/app/activity/function/hashmap/HashMapActivity.kt
package vn.loitp.app.activity.function.hashmap

import android.os.Bundle
import android.view.View
import com.loitp.annotation.IsFullScreen
import com.loitp.annotation.LogTag
import com.loitp.core.base.BaseFontActivity
import com.loitp.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_func_hashmap.*
import vn.loitp.R

@LogTag("HashMapActivity")
@IsFullScreen(false)
class HashMapActivity : BaseFontActivity(), View.OnClickListener {
    private val map: MutableMap<String, String> = HashMap()
    private var autoKey = 0

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_func_hashmap
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.setImageResource(R.color.transparent)
            this.tvTitle?.text = HashMapActivity::class.java.simpleName
        }
        btAdd.setOnClickListener(this)
        btGetKey0.setOnClickListener(this)
        btRemoveKey0.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v) {
            btAdd -> {
                map[autoKey.toString()] = "Value $autoKey"
                printMap()
                autoKey++
            }
            btGetKey0 -> {
                val value = map[0.toString()]
                showShortInformation("Click value= $value")
            }
            btRemoveKey0 -> {
                map.remove(0.toString())
                printMap()
            }
        }
    }

    private fun printMap() {
        var s = ""
        for ((key, value) in map) {
            s += "$key -> $value"
        }
        textView.text = s.ifEmpty { "No data" }
    }
}
